src/encoded/static/components/browse/components/AboveTableControls/SelectedFilesDownloadButton.js

```diff
@@ -0,0 +1,248 @@
+                <Button {...btnProps} disabled={isDisabled} onClick={this.showModal}>
+                    { children }
+                </Button>
+                { this.state.modalOpen ?
+                    <SelectedFilesDownloadModal {...{ selectedFiles, filenamePrefix, selectedFilesUniqueCount }} onHide={this.hideModal}/>
+                : null }
+            </React.Fragment>
+        );
+    }
+}
+
+
+
+class SelectedFilesDownloadModal extends React.PureComponent {
+
+    static findUnpublishedFiles = memoize(function(selectedFiles){
+        return _.any(_.values(selectedFiles || {}), function(file){
+            return !(
+                (file.from_experiment && file.from_experiment.from_experiment_set && file.from_experiment.from_experiment_set.produced_in_pub)
+                || (file.from_experiment_set && file.from_experiment_set.produced_in_pub)
+            );
+        });
+    });
+
+    constructor(props){
+        super(props);
+        this.handleAcceptDisclaimer = this.handleAcceptDisclaimer.bind(this);
+        this.state = {
+            'disclaimerAccepted' : false
+        };
+    }
+
+    handleAcceptDisclaimer(){
+        this.setState({ 'disclaimerAccepted' : true });
+    }
+
+    render(){
+        const { onHide, filenamePrefix, selectedFiles, selectedFilesUniqueCount } = this.props;
+        const { disclaimerAccepted } = this.state;
+
+        const suggestedFilename = filenamePrefix + DateUtility.display(moment().utc(), 'date-time-file', '-', false) + '.tsv';
+        const userInfo = JWT.getUserInfo();
+        const isSignedIn = !!(userInfo && userInfo.details && userInfo.details.email && userInfo.id_token);
+        const profileHref = (isSignedIn && userInfo.user_actions && _.findWhere(userInfo.user_actions, { 'id' : 'profile' }).href) || '/me';
+        const countSelectedFiles = selectedFilesUniqueCount || SelectedFilesDownloadButton.totalSelectedFilesCount(selectedFiles);
+        const foundUnpublishedFiles = SelectedFilesDownloadModal.findUnpublishedFiles(selectedFiles);
+
+        return (
+            <Modal show className="batch-files-download-modal" onHide={onHide} bsSize="large">
+
+                <Modal.Header closeButton>
+                    <Modal.Title>
+                        <span className="text-400">Download <span className="text-600">{ countSelectedFiles }</span> Files</span>
+                    </Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>
+
+                    <p>Please press the &quot;Download&quot; button below to save the metadata TSV file which contains download URLs and other information for the selected files.</p>
+                    <p>Once you have saved the metadata TSV, you may download the files on any machine or server with the following cURL command:</p>
+                    <ModalCodeSnippet filename={suggestedFilename} isSignedIn={isSignedIn} />
+
+                    <div className="extra-notes-section">
```

Comment:
Previously this entire section was getting excluded if `!foundUnpublishedFiles` so we moved most things back into a single component for speed/simplicity.